src/pg/connection.cr

```diff
@@ -41,6 +41,24 @@ module PG
       @connection.notification_handler = on_notification_proc
     end
 
+    # `Time::Location.load` doesn't do any caching, so we cache it here to avoid
+    # a time-zone lookup on every call to `time_zone`.
+    @@location_cache = Hash(String, Time::Location).new do |cache, zone_name|
+      Time::Location.load(zone_name)
+    end
+
+    def time_zone
+      if zone_name = @connection.server_parameters["TimeZone"]?
+        @@location_cache[zone_name]
+      else
+        Time::Location::UTC
+      end
+    end
+
+    def time_zone=(location : Time::Location)
+      exec "SET TIME ZONE '#{location.name}'"
+    end
```

Comment:
I don't know if this method is necessary since I removed the `property` in https://github.com/will/crystal-pg/pull/266/commits/020d847bc5da15503f740e86809ec9455c6da165 to make it automatically handle `SET TIME ZONE` queries.